src/main/java/org/mskcc/cmo/metadb/service/impl/CmoLabelGeneratorServiceImpl.java

```diff
@@ -199,6 +205,9 @@ public String generateCmoSampleLabel(SampleMetadata sampleMetadata,
 
         // get next incremement value for cmo sample counter
         Integer nextSampleCounter = getNextSampleIncrement(existingSamples);
+        if (foundSampleInExistingSamples(sampleMetadata.getPrimaryId(), existingSamples)) {
+            nextSampleCounter = nextSampleCounter - 1;
```

Comment:
same comment - we're not decreasing the counter, just using the matching counter

Reply:
When igoId matches, the counter should be the same right? Because `getNextSampleIncrement` returns matchedCounter + 1